### PR TITLE
Made some initial changes to test new styles

### PR DESCRIPTION
**Pull request by halindrome**
Simple changes to load the new stylesheets and
implement some basic changes that make them work.

To enable in your respec document, set

    useExperimentalStyles: true,

in your Respec Configuration area.

In response to issue #520

### CODE REVIEW

File: js/core/structure.js [hunk 1]
```diff
@@ -97,13 +97,19 @@ define(
                 if (!conf.noTOC) {
                     var $ul = makeTOCAtLevel($("body", doc), doc, [0], 1, conf);
                     if (!$ul) return;
-                    var $sec = $("<section id='toc'/>").append("<h2 class='introductory'>" + conf.l10n.toc + "</h2>")
+                    var w = ( conf.useExperimentalStyles ? "nav" : "section" );
```
Comment (marcoscaceres) [inline js/core/structure.js]: Maybe just:
```js
var w = ( conf.useExperimentalStyles ) ? "nav" : "section" ;
```

Comment (marcoscaceres) [inline js/core/structure.js]: (updated code above)

File: js/core/structure.js [hunk 2]
```diff
@@ -97,13 +97,19 @@ define(
...
+                    var $sec = $("<" + w + " id='toc'/>").append("<h2 class='introductory'>" + conf.l10n.toc + "</h2>")
```
Comment (marcoscaceres) [inline js/core/structure.js]: Nit (though I would prefer we only indent with 2 spaces... will eventually fix ReSpec's horrible whitespace conventions):
```JS
                    var $sec = $("<" + w + " id='toc'/>")
                                                        .append("<h2 class='introductory'>" +  conf.l10n.toc + "</h2>")
                                                        .append($ul)
```

File: js/w3c/headers.js [hunk 3]
```diff
@@ -535,6 +535,12 @@ define(
                     msg.pub("error", "ReSpec does not support automated SotD generation for TAG findings, " +
                                      "please specify one using a <code><section></code> element with ID=sotd.");
                 }
+                // experimental style info
+                if (conf.useExperimentalStyles) {
+                    $("html>head").prepend($('<meta name="viewport" content="width=device-width, initial-scale=1, width = content-width, initial-scale = 1.0, shrink-to-fit=no">'));
```
Comment (marcoscaceres) [inline js/w3c/headers.js]: I can't remember exactly, but adding this after the fact might not actually work on mobile. 
Also, should probably check if: 
```
var hasViewportMeta = doc.head.querySelector("meta[name=viewport]") ? true : false;
if (conf.useExperimentalStyles && !hasViewportMeta) { ...}
```
Before adding another on - as having two might result in strangeness.

**Comment by marcoscaceres** [comment]
Some nits, LGTM tho.

**Comment by plehegar** [comment]
We can't merge due to conflicts:

js/w3c/aria.js:
<<<<<<< HEAD
                    var toc = conf.useExperimentalStyles ? $('nav#toc', doc) : $('section#toc', doc);
                    toc.attr('role', 'navigation') ;
=======
                    var toc = $('section#toc', doc).find("ul:first");
                    toc.attr('role', 'directory') ;
>>>>>>> develop

tests/spec/core/structure-spec.js
<<<<<<< HEAD
            expect($toc.attr('role')).toEqual('navigation');
=======
            expect($toc.find("ul:first").attr('role')).toEqual('directory');
>>>>>>> develop

@halindrome any chance you can resolve them and merge?

**Comment by halindrome** [comment]
Sure. I will look when I get home.

**Comment by halindrome** [comment]
This is taken care of and I pushed a new version.  I am going to open a new
issue and tag you with an idea I havea.

On Fri, Nov 20, 2015 at 2:16 PM, Shane McCarron <halindrome@gmail.com>
wrote:

> Sure. I will look when I get home.
> On Nov 20, 2015 9:43 AM, "Philippe Le Hegaret" <notifications@github.com>
> wrote:
>
>> We can't merge due to conflicts:
>>
>> js/w3c/aria.js:
>> <<<<<<< HEAD
>> var toc = conf.useExperimentalStyles ? $('nav#toc', doc) :
>> $('section#toc', doc);
>> toc.attr('role', 'navigation') ;
>>
>>                 var toc = $('section#toc', doc).find("ul:first");
>>                 toc.attr('role', 'directory') ;
>>
>> develop
>>
>> tests/spec/core/structure-spec.js
>> <<<<<<< HEAD
>> expect($toc.attr('role')).toEqual('navigation');
>>
>>         expect($toc.find("ul:first").attr('role')).toEqual('directory');
>>
>> develop
>>
>> @halindrome <https://github.com/halindrome> any chance you can resolve
>> them and merge?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/w3c/respec/pull/522#issuecomment-158436955>.
>>
>


-- 
Shane McCarron
halindrome@gmail.com

File: js/w3c/style.js [hunk 4]
```diff
@@ -40,6 +40,9 @@ define(
                         .appendTo($("head"))
                         ;
                 }
+                else if (conf.useExperimentalStyles) {
+                    css = "http://fantasai.inkedblade.net/style/design/w3c-restyle/2016/W3C-" + statStyle;
```
Comment (marcoscaceres) [inline js/w3c/style.js]: D'oh. Can't believe I didn't see this :(

Comment (halindrome) [inline js/w3c/style.js]: Well - that's where the stylesheets are.  Should we have done something
else?

On Wed, Nov 25, 2015 at 7:55 PM, Marcos Caceres <notifications@github.com>
wrote:

> In js/w3c/style.js
> <https://github.com/w3c/respec/pull/522#discussion_r45938038>:
>
> > @@ -40,6 +40,9 @@ define(
> >                          .appendTo($("head"))
> >                          ;
> >                  }
> > +                else if (conf.useExperimentalStyles) {
> > +                    css = "http://fantasai.inkedblade.net/style/design/w3c-restyle/2016/W3C-" + statStyle;
>
> D'oh. Can't believe I didn't see this :(
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/w3c/respec/pull/522/files#r45938038>.
>



-- 
Shane McCarron
halindrome@gmail.com

Comment (marcoscaceres) [inline js/w3c/style.js]: > On 27 Nov 2015, at 3:14 AM, Shane McCarron <notifications@github.com> wrote:
> 
> In js/w3c/style.js:
> 
> > @@ -40,6 +40,9 @@ define(
> >                          .appendTo($("head"))
> >                          ;
> >                  }
> > +                else if (conf.useExperimentalStyles) {
> > +                    css = "http://fantasai.inkedblade.net/style/design/w3c-restyle/2016/W3C-" + statStyle;
> Well - that's where the stylesheets are. Should we have done something else?

Yeah, we should have stopped and complained that http is unacceptable in 2015. We should make that a rule going forward.

The project is hosted on Github, so we could ask @fantasai to just serve it from gh-pages (it probably already is, but too lazy to check from my phone:))

**Comment by plehegar** [comment]
Can we close this one now?

**Comment by plehegar** [comment]
oops nevermind.

**Pull request merged**